generate_qt_mappings.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+"""
+This script generated the Qt mapping file according to given Qt include
+directory
+
+Example usage :
+
+   $ ./generate_qt_mappings.py /usr/include/x86_64-linux-gnu/qt5 qt5_11.imp
+
+"""
+
+import argparse
+import glob
+import os
+import re
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("qt_include_dir", help="Qt include directoy")
+    parser.add_argument("output_file", help="Generated output mapping file")
+    parser.add_argument("--warnings", action="store_true", help="Display possible warning messages")
+    args = parser.parse_args()
+
+    headers = glob.glob(os.path.join(args.qt_include_dir, '**/*[!\\.h]'))
+    symbols_map = []
+    includes_map = []
+
+    class2includes = {}
+    include2classes = {}
+    for header in headers:
+
+        if os.path.isdir(header):
+            continue
+
+        class_name = os.path.basename(header)
+        module_name = os.path.basename(os.path.dirname(header))
+
+        symbols_map += [f'{{ symbol: [ "{class_name}", "private", '
```

Comment:
Rather than building a long list of strings, could you build a list of dicts that you can then serialize to json?

Something like:

    # With quoting around the class_name
    symbols_map.append({'symbol': [class_name, "private", class_name, "public"]})

And:

    json.dump(symbols_map, f, indent=2)

That way you could also add includes and symbols to the same list, or if you want to separate them, concat the lists at the end (much like you're doing now).

Reply:
And it might be worth adding a `def symbol_record(class_name)` function that generates that dict with the right quoting, etc.

Reply:
Hmm. Why is the mapping from symbol to symbol? Shouldn't it be symbol to quoted-include? Ah, I see, you're back-generating the quoted-include from the class name. Hmm. Maybe that's the best we can do :)

Reply:
> json.dump(symbols_map, f, indent=2)

Using json dump produce a more indented (and less readable?) result 

```json
{
    "symbol": [
      "QComputeCommand",
      "private",
      "<QComputeCommand>",
      "public"
    ]
  },
  {
    "include": [
      "\"@[\\\"<](Qt3DRender/)?qcomputecommand\\\\.h[\\\">]\"",
      "private",
      "<QComputeCommand>",
      "public"
    ]
  }
```
Are you sure you still want to use json module to write the symbols and includes records?

Reply:
> Why is the mapping from symbol to symbol? 

To be honest I try to make the symbol mapping the closer from the original qt5_4.imp file without really understanding what it implies :) Remember that my initial goal was to generate a mapping for include file who deals with quotes and <>